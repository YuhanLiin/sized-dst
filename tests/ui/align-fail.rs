use sized_dst::*;

#[repr(align(32))]
#[derive(Debug)]
struct Align32(u32);

fn main() {
    SizedDstA8::<dyn std::fmt::Debug, 32>::new(Align32(1));
}
