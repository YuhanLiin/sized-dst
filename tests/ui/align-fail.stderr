error[E0080]: evaluation of `sized_dst::assert::const_assert::<Align32, sized_dst::A8, 32>::{constant#1}` failed
 --> src/assert.rs
  |
  | /         assert!(
  | |             align_of::<T>() <= align_of::<A>(),
  | |             "Value alignment requirements must not exceed that of the stack object"
  | |         );
  | |_________^ the evaluated program panicked at 'Value alignment requirements must not exceed that of the stack object', $DIR/src/assert.rs:11:9
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
  --> src/assert.rs
   |
   | /     const {
   | |         assert!(
   | |             align_of::<T>() <= align_of::<A>(),
   | |             "Value alignment requirements must not exceed that of the stack object"
   | |         );
   | |     }
   | |_____^

note: the above error was encountered while instantiating `fn sized_dst::assert::const_assert::<Align32, sized_dst::A8, 32>`
   --> src/lib.rs
    |
    |         assert::const_assert::<T, A, N>();
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
